### ISSUE TITLE
CmdStanPy hangs but pystan engine works with same input from fbprophet

### ISSUE THREAD

**Issue by varunsinghal** [facebook/prophet#1605]
Hi All

I am facing this issue with fbprophet, where I have two virtual environments one with Pystan installed and in other one cmdstanpy installed. On running the command, the cmdstanpy (== 0.4 ) hangs at sampling.

` start chain 1
2020-07-30 22:54:44,420  - DEBUG - cmdstanpy:496 - sampling: C:\Users\<USER>\<APP NAME>\venv\lib\site-packages\fbprophet\stan_model\prophet_model.bin random seed=90476 data file=C:\Users\<USER>\AppData\Local\Temp\tmp0_oxojs5\tmppn_7lj45.json init=C:\Users\<USER>\AppData\Local\Temp\tmp0_oxojs5\tmpa4kwrjnp.json output file=C:\Users\<USER>\AppData\Local\Temp\tmp0_oxojs5\stan-prophet_model-draws-1-4k6vyokx.csv method=optimize algorithm=newton iter=10000`

The model I am requesting is this.
`Prophet(growth='linear',
                             changepoint_prior_scale=<scale_value>,
                             interval_width=<interval_width>,
                             changepoint_range=1)`

Can someone please help me out to understand why there is a difference in behavior of fbprophet for two different engines (pystan and cmdstanpy).

The same code is running with pystan as backend engine. Is there any change in parameters I can make to make the behavior same?

Thanks.

**Comment by bletham** [comment]
The Stan model is the same for both. The only difference would be the differences in the APIs for each backend (you can see the code for it here: https://github.com/facebook/prophet/blob/master/python/fbprophet/models.py ). 

0.4 is a fairly old version of cmdstanpy, I'd recommend installing the latest fbprophet master branch here at the github which is pinned to cmdstanpy==0.9.5 and see if that helps. Otherwise, could you verify that cmdstanpy is working with a simple model from their docs?

**Comment by varunsinghal** [comment]
Hi @bletham 

Thanks for your comments. I will try the latest copy of cmdstanpy 0.9.5.

In the meanwhile, I would like to show you the logs generated for the below model for both pystan and cmdstanpy

`Prophet(growth='linear', changepoint_prior_scale=<scale_value>, interval_width=<interval_width>, changepoint_range=1)`

Pystan

```Initial log joint probability = -77.1904
Iteration  1. Log joint probability =    4.16921. Improved by 81.3596.
Iteration  2. Log joint probability =    15.1475. Improved by 10.9783.
Iteration  3. Log joint probability =     96.423. Improved by 81.2755.
Iteration  4. Log joint probability =     127.91. Improved by 31.487.
Iteration  5. Log joint probability =    160.943. Improved by 33.0333.
Iteration  6. Log joint probability =    172.829. Improved by 11.8861.
```

Cmdstanpy

```
start chain 1 
2020-07-30 22:54:44,420 - DEBUG - cmdstanpy:496 - sampling: C:\Users\<USER>\<APP NAME>\venv\lib\site-packages\fbprophet\stan_model\prophet_model.bin random seed=90476 data file=C:\Users\<USER>\AppData\Local\Temp\tmp0_oxojs5\tmppn_7lj45.json init=C:\Users\<USER>\AppData\Local\Temp\tmp0_oxojs5\tmpa4kwrjnp.json output file=C:\Users\<USER>\AppData\Local\Temp\tmp0_oxojs5\stan-prophet_model-draws-1-4k6vyokx.csv method=optimize algorithm=newton iter=10000
```

**Comment by varunsinghal** [comment]
@bletham 
Thanks, upgrade to master branch worked for me.
